lib/encode.js

```diff
@@ -63,21 +63,7 @@ encode.string = function (buffers, data) {
 }
 
 encode.number = function (buffers, data) {
-  var maxLo = 0x80000000
-  var hi = (data / maxLo) << 0
-  var lo = (data % maxLo) << 0
-  var val = hi * maxLo + lo
-
-  buffers.push(Buffer.from('i' + val + 'e'))
-
-  if (val !== data && !encode._floatConversionDetected) {
-    encode._floatConversionDetected = true
-    console.warn(
-      'WARNING: Possible data corruption detected with value "' + data + '":',
-      'Bencoding only defines support for integers, value was converted to "' + val + '"'
-    )
-    console.trace()
-  }
+  buffers.push(Buffer.from('i' + BigInt(data) + 'e'))
```

Comment:
This is of course a breaking change, as `BigInt()` will through if you hand it a float or anything else invalid, which also made the float conversion warning obsolete

Reply:
I think I am fine with a breaking change, especially if we reach support for the full v2 spec for that. 
And I did not know that BitInt is part of the core now, that neat!